modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/dsl/AlertMessage.scala

```diff
@@ -0,0 +1,107 @@
+        case Some(cause) => go(msg :: acc, cause)
+        case None => (msg :: acc).reverse.mkString(": ")
+      }
+    }
+
+    Show.show(go(Nil, _))
+  }
+
+  /** Restrict the length of an alert message to be compliant with alert iglu schema */
+  private val MaxAlertPayloadLength = 4096
+
+  case class FolderIsAlreadyLoaded(folder: BlobStorage.Folder) extends AlertMessage
+  case class FolderIsUnloaded(folder: BlobStorage.Folder) extends AlertMessage
+  case class ShreddingIncomplete(folder: BlobStorage.Folder) extends AlertMessage
+  case object UnloadedFoldersDetected extends AlertMessage
+  case class RetryableLoadFailure(folder: BlobStorage.Folder, cause: Throwable) extends AlertMessage
+  case class TerminalLoadFailure(folder: BlobStorage.Folder, cause: Throwable) extends AlertMessage
+  case class FailedInitialConnection(cause: Throwable) extends AlertMessage
+  case class FailedToCreateEventsTable(cause: Throwable) extends AlertMessage
+  case class FailedToCreateManifestTable(cause: Throwable) extends AlertMessage
+  case class FailedHealthCheck(cause: Throwable) extends AlertMessage
+  case class FailedFolderMonitoring(cause: Throwable, numFailures: Int) extends AlertMessage
+
+  val getFolder: AlertMessage => Option[BlobStorage.Folder] = {
+    // folder-specific messages
+    case FolderIsAlreadyLoaded(folder) => Some(folder)
+    case FolderIsUnloaded(folder) => Some(folder)
+    case ShreddingIncomplete(folder) => Some(folder)
+    case RetryableLoadFailure(folder, _) => Some(folder)
+    case TerminalLoadFailure(folder, _) => Some(folder)
+    // general messages
+    case UnloadedFoldersDetected => None
+    case FailedInitialConnection(_) => None
+    case FailedToCreateEventsTable(_) => None
+    case FailedToCreateManifestTable(_) => None
+    case FailedHealthCheck(_) => None
+    case FailedFolderMonitoring(_, _) => None
+  }
+
+  def getSeverity: AlertMessage => Severity = {
+    // information: no action required to fix the problem
+    case FolderIsAlreadyLoaded(_) => Severity.Info
+    case RetryableLoadFailure(_, _) => Severity.Info
+    // warnings: require external intervention to fix the problem
+    case TerminalLoadFailure(_, _) => Severity.Warning
+    case UnloadedFoldersDetected => Severity.Warning
+    case FailedFolderMonitoring(_, _) => Severity.Warning
+    case FailedHealthCheck(_) => Severity.Warning
+    // These warnings might be downgraded to "info" in a future release because the
+    // `UnloadedFoldersDetected` is already a call to action:
+    case ShreddingIncomplete(_) => Severity.Warning
+    case FolderIsUnloaded(_) => Severity.Warning
+    // errors: these prevent the loader from running whatsoever. The loader will crash and exit after sending these alerts
+    case FailedInitialConnection(_) => Severity.Error
+    case FailedToCreateEventsTable(_) => Severity.Error
+    case FailedToCreateManifestTable(_) => Severity.Error
+  }
+
+  def getString(am: AlertMessage): String = {
```

Comment:
How about we rename `AlertMessage` to simple `Alert` and this  function to `getMessage`?